### PR TITLE
Tales of the Past Machinima Easter Egg Characters

### PR DESCRIPTION
**Pull request by Malicos**
## Changelog:
- Added the cast of Tales of the Past machinima trilogy:
# EDEN AURORAE GUILD
   - Blazer the Rogue (unlanded human adventurer, waiting for RTP to implement as playable landless)
   - Yimo the Mage (unlanded gnome adventurer, waiting for RTP to implement as playable landless)
# THE ANCIENTS GUILD
   - Ball the Warrior (unlanded Tauren mercenary/adventurer, waiting for RTP to implement as playable landless)
   - Jubai the Ancient (Druid) (unlanded Tauren mercenary/adventurer, waiting for RTP to implement as playable landless)
# MISCELLANEOUS
   - Serphentos (Dwarven warrior in Magni's court)
   - Monóri Silverleaf (Night Elf sentinel under Nythalias Silverleaf, also becomes a Dark Ranger circa 609-610)
      - Naelorn and Nythalias Silverleaf added as Monóri's parents based on https://chronicles-of-lordaeron.fandom.com/wiki/Monori
      - Made Nendis a Night Elven territory as per lore. The Silverleafs hail from there and the duchy is held by Nythalias in all startdates. https://warcraft.wiki.gg/wiki/Nendis
   - Thainor (human necromancer in the court of Ner'zhul from 593 KC)
   - Gwaar (Tauren mystic under Rerel'man Cliffwalker)

## TODO:
- [ ] Implement playable landless characters properly
- [x] DNAs (with exception of Tauren which use static models)
- [x] Balancing

## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:
- Use Find Character to locate all entries and check if they spawn properly.

### CODE REVIEW

**Review by edmund-h — CHANGES_REQUESTED** [review]
please annotate any magic paths you would like the characters to use and what level (1-5). otherwise LGTM

File: history/characters/wc_easteregg_characters.txt [hunk 1]
```diff
@@ -457,4 +457,271 @@ easteregg13={ #Shrek reference
...
+	name = "Blazer"
+	dna = blazer_dna
+	culture = azerothian
+	religion = holy_light
+	disallow_random_traits = yes
+	#Stats
+	diplomacy = 4
+	martial = 7
+	stewardship = 4
+	intrigue = 15
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: 15 intrigue? Isn't that overpowered?

File: history/characters/wc_easteregg_characters.txt [hunk 2]
```diff
@@ -457,4 +457,271 @@ easteregg13={ #Shrek reference
...
+easteregg19 = {
+	name = "Monóri"
+	#dna = monori_dna
+	dynasty = easteregg_dyn11 # Silverleaf
+	culture = night_elf
+	religion = kaldorei_religion
+	disallow_random_traits = yes
+	#Stats
+	diplomacy = 10
+	martial = 15
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: Looks overpowered

File: history/characters/wc_easteregg_characters.txt [hunk 3]
```diff
@@ -457,4 +457,271 @@ easteregg13={ #Shrek reference
...
+	stewardship = 5
+	intrigue = 3
+	learning = 4
+	prowess = 25
+	#Personality
+	trait = compassionate
+	trait = generous
+	trait = diligent
+	#Education
+	trait = education_martial_5
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: I think level 5 martial is too much for an easter egg character, especially considering that most lore characters like Maiev and Broll Bearmantle don't exceed level 4.

File: history/characters/wc_easteregg_characters.txt [hunk 4]
```diff
@@ -457,4 +457,271 @@ easteregg13={ #Shrek reference
...
+	#dna = gwaar_dna
+	culture = tauren
+	religion = earth_mother_worship
+	disallow_random_traits = yes
+	#Stats
+	diplomacy = 6
+	martial = 4
+	stewardship = 4
+	intrigue = 5
+	learning = 20
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: 20 base learning is massively overpowered.

File: history/characters/wc_easteregg_characters.txt (hunk 1 above)
Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: rebalanced

File: history/characters/wc_easteregg_characters.txt (hunk 4 above)
Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: rebalanced

File: history/characters/wc_easteregg_characters.txt (hunk 3 above)
Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: rebalanced

File: history/characters/wc_easteregg_characters.txt [hunk 5]
```diff
@@ -457,4 +457,267 @@ easteregg13={ #Shrek reference
...
+	intrigue = 7
+	learning = 4
+	#Personality
+	trait = brave
+	trait = gregarious
+	trait = stubborn
+	#Education
+	trait = education_intrigue_2
+	#Other
+	trait = adventurer
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt:484]: keep in mind that once Roads to Power comes out this trait will be renamed to Veteran Adventurer for those who used to be adventurers

Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt:484]: noted

File: history/characters/wc_easteregg_characters.txt [hunk 6]
```diff
@@ -457,4 +457,276 @@ easteregg13={ #Shrek reference
...
+	558.6.3 = { 
+		birth = yes
+		trait = creature_human
+	}
+	593.1.1={
+		employer = 10014		# Ner'zhul [2353]
+		religion = death_god
+	}
+	# level 3 necromantic
+	610.1.10 = { death_reason = death_battle }	# Killed in Ironforge by Serphentos
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: If he was killed by serphentos shouldn't that be reflected in the code?
Like:
```
610.1.10 = { 
   death = {
        death_reason = death_battle 
        killer = easteregg18
    }
}
```

File: history/characters/wc_easteregg_characters.txt [hunk 7]
```diff
@@ -457,4 +457,276 @@ easteregg13={ #Shrek reference
...
+	#Other
+	trait = adventurer
+	trait = intellect_good_1
+	trait = magic_good_2
+	466.5.10 = { 
+		birth = yes
+		trait = creature_gnome
+	}
+	# level 3 frost mage
+	608.10.14 = { death_reason = death_battle }	# Killed in Naxxramas
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: if it has `death_reason` then it should be:
`608.10.14 = { death = { death_reason = death_battle } }`

Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: fixed

File: history/characters/wc_easteregg_characters.txt (hunk 6 above)
Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: fixed

File: history/characters/wc_easteregg_characters.txt [hunk 8]
```diff
@@ -457,4 +457,281 @@ easteregg13={ #Shrek reference
...
+	#Other
+	trait = physique_good_2
+	trait = lifestyle_blademaster
+	377.8.24 = { 
+		birth = yes
+		trait = creature_dwarf
+	}
+	511.7.15 = {
+		employer = 17001 # Magni [4200]
+		give_council_position = councillor_marshal
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: How lore accurate is this?

Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: made Serp no longer on the council and moved Monóri to Auberdine

File: history/characters/wc_easteregg_characters.txt [hunk 9]
```diff
@@ -457,4 +457,280 @@ easteregg13={ #Shrek reference
...
+	trait = education_learning_3
+	#Other
+	trait = adventurer
+	trait = magic_good_1
+	506.3.10 = { 
+		birth = yes
+		trait = creature_tauren
+	}
+	583.1.1 = {
+		give_nickname = nick_the_ancient
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt:585]: His ingame nickname is 'Brave' for some reason.
Στιγμιότυπο οθόνης 2024-09-19 140312

Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt:585]: It seems to be given to many other Tauren for some reason and it overrides everything else

Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt:585]: also issue is not related to this branch exclusively

File: history/characters/wc_easteregg_characters.txt [hunk 10]
```diff
@@ -457,4 +457,280 @@ easteregg13={ #Shrek reference
...
+	trait = honest
+	trait = just
+	trait = diligent
+	#Education
+	trait = education_learning_3
+	#Other
+	trait = intellect_good_1
+	trait = lifestyle_mystic
+	trait = magic_good_2
+	491.2.16 = {
```
Comment (Grekopithikos) [inline history/characters/wc_easteregg_characters.txt]: Is this accurate?
In 605 he's 114 years old. Tauren have a +25 life expectancy.

File: localization/english/names/wc_character_names_l_english.yml [hunk 11]
```diff
@@ -18326,4 +18326,12 @@
...
  Tcha'Du:0 "Tcha'Du"
- Shrok:0 "Shrok"
\ No newline at end of file
+ Shrok:0 "Shrok"
+ Blazer:0 "Blazer"
+ Yimo:0 "Yimo"
+ Ball:0 "Ball"
+ Jubai:0 "Jubai"
+ Serphentos:0 "Serphentos"
+ Monóri:0 "Monóri"
```
Comment (Grekopithikos) [inline localization/english/names/wc_character_names_l_english.yml]: `[13:44:19][localization_reader.cpp:445]: Invalid character 'Γ' in key name 'MonΓ³ri' at line 18335 and column 4 in localization/english/names/wc_character_names_l_english.yml`

File: history/characters/wc_easteregg_characters.txt (hunk 10 above)
Comment (Malicos) [reply history/characters/wc_easteregg_characters.txt]: Made him younger.

File: localization/english/names/wc_character_names_l_english.yml (hunk 11 above)
Comment (Malicos) [reply localization/english/names/wc_character_names_l_english.yml]: fixed

**Review by Grekopithikos — DISMISSED** [review]
LGTM!

**Review by robmart — DISMISSED** [review]

**Review by edmund-h — APPROVED** [review]

**Pull request merged**